### PR TITLE
Complete class declaration for FormData spec

### PR DESCRIPTION
**Pull request by acusti**
Reference: https://xhr.spec.whatwg.org/#interface-formdata

### CODE REVIEW

**Comment by acusti** [comment]
I manually updated the line numbers in the expectations in `tests/fetch/fetch.exp` to match the new line numbers form the expanded FormData declaration. Those tests are now passing. Got an error from `_build/src/parser/test/run_hardcoded_tests.native src/parser/test/hardcoded_tests.js` however:

```
cd /home/travis/build/facebook/flow/src/parser//../..; \
	_build/src/parser/test/run_hardcoded_tests.native src/parser/test/hardcoded_tests.js
Fatal error: exception Not_found
```

Travis job: https://travis-ci.org/facebook/flow/jobs/174823776

Don’t know why that would be, but I’d be happy to take a look at it if someone could point me in the right direction.

**Comment by acusti** [comment]
Also, if you’d prefer not to include the entire `FormData` interface as defined by the spec because it isn’t supported by all major browsers, we can close this PR.

File: lib/bom.js [hunk 1]
```diff
@@ -241,7 +241,11 @@ declare class DOMParser {
 }
 
 declare class FormData {
-    append(name: any, value: any, blobName?: string): void;
+    has(name: string): boolean;
+    get(name: string): ?string;
```
Comment (gabelevi) [inline lib/bom.js]: Looking at the spec, I think the return type is `?(string | File)`

File: lib/bom.js [hunk 2]
```diff
@@ -241,7 +241,11 @@ declare class DOMParser {
...
+    set(name: string, value: any, blobName?: string): void;
```
Comment (gabelevi) [inline lib/bom.js]: An overload would also work

```
set(name: string, value: string): void
set(name: string, value: Blob, filename?: string): void
```

File: lib/bom.js [hunk 3]
```diff
@@ -241,7 +241,11 @@ declare class DOMParser {
...
+    append(name: string, value: any, blobName?: string): void;
```
Comment (gabelevi) [inline lib/bom.js]: ditto for append

**Comment by gabelevi** [comment]
I'm cool including this! And sorry for the delay! I have some notes, but since this is so delayed I'll address them myself :)

**Comment by gabelevi** [comment]
Well I feel silly. Went to rebase this branch and found out we already merged a PR with definitions for this (https://github.com/facebook/flow/pull/2471) 9 days ago. So closing this out :)

Thanks again and sorry for the trouble!

**Comment by acusti** [comment]
Hah, that’s pretty funny! Glad to hear the definitions made it in, and thanks for the work and feedback.

**Pull request closed without merging**